src/Document.js

```diff
@@ -0,0 +1,67 @@
+const {HTMLElement, HTMLLoadableElement, Node} = require('./DOM');
+const symbols = require('./symbols');
+
+class Document extends HTMLLoadableElement {
+  constructor() {
+    super('DOCUMENT');
+
+    this.hidden = false;
+  }
+
+  get nodeType() {
+    return Node.DOCUMENT_NODE;
+  }
+
+  get pointerLockElement() {
+    if (this.defaultView.top === this.defaultView) {
+      return this[symbols.pointerLockElementSymbol];
+    } else {
+      return this.defaultView.top.document.pointerLockElement;
+    }
+  }
+  set pointerLockElement(pointerLockElement) {}
+  get fullscreenElement() {
+    if (this.defaultView.top === this.defaultView) {
+      return this[symbols.fullscreenElementSymbol];
+    } else {
+      return this.defaultView.top.document.fullscreenElement;
+    }
+  }
+  set fullscreenElement(fullscreenElement) {}
+
+  exitPointerLock() {
+    const topDocument = this.defaultView.top.document;
+
+    if (topDocument[symbols.pointerLockElementSymbol] !== null) {
+      topDocument[symbols.pointerLockElementSymbol] = null;
+
+      process.nextTick(() => {
+        topDocument._emit('pointerlockchange');
+      });
+    }
+  }
+  exitFullscreen() {
+    const topDocument = this.defaultView.top.document;
+
+    if (topDocument[symbols.fullscreenElementSymbol] !== null) {
+      topDocument[symbols.fullscreenElementSymbol] = null;
+
+      process.nextTick(() => {
+        topDocument._emit('fullscreenchange');
+      });
+    }
+  }
+}
+global.Document = Document;
```

Comment:
Breaking apart core.js makes it easier for this type of bug to slip through.

Reply:
Yeah, it's a temporary step. I'm using `global` until we can break the rest of `core.js` into modules because there would be circular dependencies in the meantime, and I wanted to do it piecewise.